From f71157c13a872c8f954fa1996c44d67cfb1259b9 Mon Sep 17 00:00:00 2001
From: Rexiel Scarlet <37258415+Rexcrazy804@users.noreply.github.com>
Date: Sun, 2 Aug 2026 00:45:32 +0400
Subject: [PATCH] added jj support to hydro

fish broke something that broke hydro git branch regex
so I fix it by replacing git with jj changeid :)
---
 conf.d/hydro.fish | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/conf.d/hydro.fish b/conf.d/hydro.fish
index fad292e..2ea84cf 100644
--- a/conf.d/hydro.fish
+++ b/conf.d/hydro.fish
@@ -67,13 +67,10 @@ function _hydro_prompt --on-event fish_prompt
 
     fish --private --command "
         set branch (
-            command git branch --show-current 2>/dev/null ||
-            command git describe --tags --exact-match HEAD 2>/dev/null ||
-            command git rev-parse --short HEAD 2>/dev/null |
-                string replace --regex -- '(.+)' '@\$1'
+            command jj log -r @ --no-graph --template 'change_id.short(8)' 2>/dev/null
         )
 
-        test -z \"\$$_hydro_git\" && set --universal $_hydro_git \"\$branch \"
+        set --universal $_hydro_git \"\$branch \"
 
         command git diff-index --quiet HEAD 2>/dev/null
         test \$status -eq 1 ||
-- 
2.55.0

